core/templates/pages/topic-editor-page/navbar/topic-editor-navbar.component.ts

```diff
@@ -91,6 +93,38 @@ export class TopicEditorNavbarComponent
     this.prepublishValidationIssues = prepublishTopicValidationIssues.concat(
       subtopicPrepublishValidationIssues
     );
+    this.checkSupersedingSkills().then(supersedingIssues => {
+      this.validationIssues = this.validationIssues.concat(supersedingIssues);
+    });
+  }
+
+  async checkSupersedingSkills(): Promise<string[]> {
+    let skillIDsToBeChecked = this.topic.getSkillIds();
+    const promises = skillIDsToBeChecked.map(async skillId => {
+      return this.skillBackendApiService
+        .fetchSkillAsync(skillId)
```

Comment:
This is not great, it performs a backend call inside a loop.. Note that topic validation may happen multiple times and making so many backend calls is inefficient.

Why can't we fetch what is required at the start and do the necessary validations when required?